[Feat #9] 관리자 API 일부 구현

jiminh00 opened the pull request:
## 작업 내용
관리자 기능 일부
- 사용자 조회 API 구현 
- 사용자 삭제 API 구현
- 사용자 로그 API 구현

## 관련 이슈
close #9 

## 참고
- `Spring Security Filter Chain`에 role ADMIN 검사를 추가하기 위해 `CustomUserDetails` 클래스를 일부 수정하였습니다. Spring Security 사용이 아직 익숙하지 않아 이 부분 중심으로 확인해주시면 감사하겠습니다. 🙇‍♀️
- 테스트 코드를 일부 작성하였습니다. 참고해주세요.

jiminh00 committed:
[Feat] #9 관리자 - 사용자 리스트 조회 API 구현

- SecurityConfig에 /admin/** 경로에 대해 hasRole("ADMIN") 검사 추가
- CustomUserDetails에서 GrantedAuthority 생성 시 "ROLE_" 접두사 추가하여 ROLE 검증 로직 개선
- 사용자 리스트 조회 API 구현 완료

jiminh00 committed:
[Feat] #9 관리자 - 사용자 삭제 API 구현

- 사용자 삭제 API 구현 완료
(일치하는 사용자 없을 경우 예외 발생)

jiminh00 committed:
[Feat] #9 관리자 - 로그 조회 API 구현

- 로그 조회 API 구현

jiminh00 committed:
[Feat] #9 관리자 기능 Controller 테스트

- 관리자 유저의 API 호출 가능 여부 확인
- 존재하지 않은 유저 삭제 시 404 응답 여부 확인

jiminh00 committed:
[Feat] #9 관리자 기능 Service 테스트

- 로그 조회 응답 DTO 형식대로 반환하는지 확인

coffeesigma commented on jobbotdari_user/src/main/java/com/team9/jobbotdari/entity/Log.java:
```diff
@@ -29,5 +30,19 @@ public class Log {
+        String formattedDate = (createdAt != null) ? createdAt.format(formatter) : "null";
+
+        return String.format(
+                "%s | userId=%s, action=%s, description=%s",
```
로그 출력 형식이 좀만 더 이쁘면 좋을거같아요. 아니면 String list가 아니라 프론트에서 꾸밀 수 있게 {userId: blabla, action: blabla, description: blabla} 형식으로 하는 방법?

jiminh00 replied:
말씀하신 것처럼 Log를 String으로 변환하지 않고, 구성요소를 하나씩 보내는 게 나을 것 같네요. 수정해보겠습니다. 감사합니다.

coffeesigma reviewed (approved):
로그 추가 기능과 유저 삭제 관련 기능이 잘 구현된 것 같습니다.
고생하셨습니다!

jiminh00 committed:
[Feat] #9 로그 API 응답 DTO 변경

- 기존 사용자별 그룹화된 로그 반환 방식에서 단일 로그 항목 반환으로 변경
- LogListResponseDto에 userId, name, description, action, createdAt 필드 추가

coffeesigma reviewed (approved):
코드가 더 간단해졌네요! 좋슴다

coffeesigma committed:
Merge branch 'main' into feat/#9
